packages/x-date-pickers/src/AdapterDayjs/AdapterDayjs.ts

```diff
@@ -73,6 +75,7 @@ const defaultFormats: AdapterFormats = {
   month: 'MMMM',
   monthShort: 'MMM',
   dayOfMonth: 'D',
+  fullDayOfMonth: 'Do',
```

Comment:
Used to generate friendly `aria-valuetext`, i.e. `31st`.
If we feel that this is too heavy, we can keep just the number.

Reply:
This will be overridable using `<LocaizationProvider dateFormats={{ }}>`
If we are fine with it, then I think we are good

Reply:
Well, other formats are overridable as well. 🤔 
I've initially made this format internal, but later thought that it makes little to no sense... 🤷

Reply:
I agree with you, it's fine here :ok